### PR TITLE
Refs #32901 - updates for Katello 4.0 where pulp2 is no longer supported

### PR DESCRIPTION
**Pull request by jjeffers**
This PR addresses a bad CP into the origin 4.0.2 release. Without it, CVV incremental updates will not work.

### CODE REVIEW

File: app/lib/actions/katello/content_view_version/incremental_update.rb [hunk 1]
```diff
@@ -169,9 +168,6 @@ def copy_repos(new_repo, new_version, content, dep_solve)
             copy_output += copy_yum_content(new_repo, solve_dependencies,
                                             content[:package_ids],
                                             content[:errata_ids])
-
-            plan_action(Katello::Repository::MetadataGenerate, new_repo)
```
Comment (ianballou) [inline app/lib/actions/katello/content_view_version/incremental_update.rb:173]: Why are these removed here? I believe they're still needed for types other than `:pulp3_yum`.

Comment (jjeffers) [reply app/lib/actions/katello/content_view_version/incremental_update.rb:173]: In 4.0, there was no call to `copy_repos`, but I can see why they would still be needed there. 

The original 9441 was picked, but there was a dependent PR that was the "pulp2 action removal" work, which never made it into 4.0.

Comment (ianballou) [reply app/lib/actions/katello/content_view_version/incremental_update.rb:173]: Does it not work to make it look like it does in master right now? I realized also that `copy_deb_content` would bug out if there are Pulp 3 deb repos to be copied.

Comment (jjeffers) [reply app/lib/actions/katello/content_view_version/incremental_update.rb:173]: No because master has diverged sufficiently from 4.0 the change set can't be the same. That was the problem with the CP as it is.

File: test/actions/katello/content_view_version_test.rb [hunk 2]
```diff
@@ -28,98 +28,6 @@ class IncrementalUpdateTest < TestBase
       katello_repositories(:fedora_17_x86_64)
     end
 
-    it 'plans' do
```
Comment (ianballou) [inline test/actions/katello/content_view_version_test.rb:31]: We kept this test in master but changed it up a little. Will matching it up with master work? https://github.com/Katello/katello/blob/master/test/actions/katello/content_view_version_test.rb#L31

**Review by ianballou — COMMENTED** [review]
Looks good, just one question about a test that was deleted.

File: test/actions/katello/content_view_version_test.rb [hunk 3]
```diff
@@ -39,87 +39,16 @@ class IncrementalUpdateTest < TestBase
...
-        :author => "author1",
-        :version => "2.2.3"
-      )
-
-      assert_raise RuntimeError, "Adding multiple versions of the same Puppet Module is not supported by incremental update.  The following Puppet Modules have duplicate versions in the incremental update content list: [\"name1-author1\"]" do
-        action.send(:check_puppet_module_duplicates, [module1.id, module2.id])
-      end
-    end
-
+
```
Comment (ianballou) [inline test/actions/katello/content_view_version_test.rb:51]: I think rubocop is unhappy with his line.

**Review by ianballou — APPROVED** [review]
Assuming the tests pass, it looks fine to me!
